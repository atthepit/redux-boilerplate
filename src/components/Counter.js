import React from 'react'

const Counter = ({ count, increment, decrement }) =>
  <div>
    <p>{count}</p>
    <button onClick={ decrement }>{'-'}</button>
    <button onClick={ increment }>{'+'}</button>
  </div>

Counter.prototype.propTypes = {
  count: React.PropTypes.number,
  increment: React.PropTypes.func,
  decrement: React.PropTypes.func,
}

export default Counter
