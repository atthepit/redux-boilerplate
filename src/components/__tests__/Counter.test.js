import React from 'react'
import Counter from '../Counter'
import renderer from 'react-test-renderer'

describe('Test Counter component', () => {
  it('renders correctly', () => {
    const count = 0
    const increment = jest.fn
    const decrement = jest.fn

    const props = {
      count,
      increment,
      decrement,
    }

    const tree = renderer.create(
      <Counter { ...props } />
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
