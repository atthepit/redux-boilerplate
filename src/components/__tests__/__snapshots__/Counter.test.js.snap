exports[`Test Counter component renders correctly 1`] = `
<div>
  <p>
    0
  </p>
  <button
    onClick={[Function]}>
    -
  </button>
  <button
    onClick={[Function]}>
    +
  </button>
</div>
`;
